source/logrepl/internal/subscription.go

```diff
@@ -315,20 +334,40 @@ func (s *Subscription) sendStandbyStatusUpdate(ctx context.Context) error {
 		return fmt.Errorf("walWrite (%s) should be >= walFlush (%s)", s.walWritten, walFlushed)
 	}
 
+	// N.B. Manage replication slot lag, by responding with the last server LSN, when
+	//      all previous slot relevant msgs have been written and flushed
+	replyWithWALEnd := walFlushed == s.walWritten && walFlushed < s.serverWALEnd
+
 	sdk.Logger(ctx).Trace().
-		Str("walWrite", s.walWritten.String()).
-		Str("walFlush", walFlushed.String()).
-		Str("walApply", walFlushed.String()).
+		Stringer("wal_write", s.walWritten).
+		Stringer("wal_flush", walFlushed).
+		Stringer("server_wal_end", s.serverWALEnd).
+		Bool("server_wal_end_sent", replyWithWALEnd).
 		Msg("sending standby status update")
 
-	err := pglogrepl.SendStandbyStatusUpdate(ctx, s.conn, pglogrepl.StandbyStatusUpdate{
+	if replyWithWALEnd {
+		if err := pglogrepl.SendStandbyStatusUpdate(ctx, s.conn, pglogrepl.StandbyStatusUpdate{
+			WALWritePosition: s.serverWALEnd,
+		}); err != nil {
+			return fmt.Errorf("failed to send standby status update with server end lsn: %w", err)
+		}
+
+		return nil
+	}
+
+	// No messages have been flushed, in this case set the flushed
```

Comment:
@lovromazgon This part is kind of tricky, since the last flushed wal on start will be zero since no acks have happened, in which case, we can only safely assume all data before the consistent point or position was flushed.

really ideally, we should be reading the slot and taking the last flushed lsn in there, but this is only applicable to cases where a) the position was entirely lost b) there is an existent slot.